lang/cs/Org.Apache.REEF.Network/Elastic/Operators/Logical/ElasticOperator.cs

```diff
@@ -0,0 +1,591 @@
+
+            if (!hasReacted && _next != null)
+            {
+                _next.OnTaskMessage(message, ref returnMessages);
+            }
+        }
+
+        /// <summary>
+        /// Add a task to the operator.
+        /// The operator must have called Build() before adding tasks.
+        /// </summary>
+        /// <param name="taskId">The id of the task to add</param>
+        /// <returns>True if the task is new and is added to the operator</returns>
+        public virtual bool AddTask(string taskId)
+        {
+            var newTask = false;
+
+            if (!_operatorFinalized)
+            {
+                throw new IllegalStateException("Operator needs to be finalized before adding tasks.");
+            }
+
+            if (!_operatorStateFinalized)
+            {
+                // If state is finalized tasks can join the topology only explicitly.
+                newTask = _topology.AddTask(taskId, _failureMachine);
+            }
+
+            if (_next != null)
+            {
+                // A task is new if it got added by at least one operator
+                return _next.AddTask(taskId) || newTask;
+            }
+
+            return newTask;
+        }
+
+        /// <summary>
+        /// Finalizes the operator.
+        /// </summary>
+        /// <returns>The same finalized operator</returns>
+        public virtual ElasticOperator Build()
+        {
+            if (_operatorFinalized)
+            {
+                throw new IllegalStateException("Operator cannot be built more than once.");
+            }
+
+            if (_prev != null)
+            {
+                _prev.Build();
+            }
+
+            _operatorFinalized = true;
+
+            return this;
+        }
+
+        /// <summary>
+        /// Finalizes the operator state. After BuildState, no more tasks can be added
+        /// to the Operator.
+        /// </summary>
+        /// <returns>The same operator with the finalized state</returns>
+        public virtual ElasticOperator BuildState()
+        {
+            if (_operatorStateFinalized)
+            {
+                throw new IllegalStateException("Operator state cannot be built more than once.");
+            }
+
+            if (!_operatorFinalized)
+            {
+                throw new IllegalStateException("Operator need to be build before finalizing its state.");
+            }
+
+            if (_next != null)
+            {
+                _next.BuildState();
+            }
+
+            _topology.Build();
+
+            LogOperatorState();
+
+            _operatorStateFinalized = true;
+
+            return this;
+        }
+
+        /// <summary>
+        /// Generate the data serializer configuration for the target operator.
+        /// </summary>
+        /// <param name="confBuilder">The conf builder where to attach the codec configuration</param>
+        internal virtual void GetCodecConfiguration(ref IConfiguration confBuilder)
```

Comment:
Ok so you are suggesting that instead of having each operator merging the current conf with its internal conf and pass the result to the next one, to have each operator returning its conf and having the previous op do the merge? (At the end the only change is who is merging what). I fixed the `confBuilder` typo.